code/datums/actions/item_action.dm

```diff
@@ -0,0 +1,33 @@
+//Presets for item actions
+/datum/action/item_action
+	name = "Item Action"
+	check_flags = AB_CHECK_HANDS_BLOCKED|AB_CHECK_CONSCIOUS
+	button_icon_state = null
+	// If you want to override the normal icon being the item
+	// then change this to an icon state
+
+/datum/action/item_action/Trigger(trigger_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(target)
+		var/obj/item/I = target
+		I.ui_action_click(owner, src)
+	return TRUE
+
+/datum/action/item_action/ApplyIcon(atom/movable/screen/movable/action_button/current_button, force)
+	var/obj/item/item_target = target
+	if(button_icon && button_icon_state)
+		// If set, use the custom icon that we set instead
+		// of the item appearence
+		..()
+	else if((target && current_button.appearance_cache != item_target.appearance) || force) //replace with /ref comparison if this is not valid.
```

Comment:
this is probably fine, but is it ok if you use else \newline if since "else if" isnt real here? 
or even something like
```
        else 
	    if(!target || current_button.appearance_cache == item_target.appearance) || !force)
	        return
	    ....
```